### PR TITLE
Fix/remove ids from auth models 3521

### PR DESCRIPTION
**Pull request by N-o-Z**

### CODE REVIEW

File: pkg/auth/model/model.go [hunk 1]
```diff
@@ -101,52 +101,35 @@ type BaseUser struct {
...
 }
 
-type User struct {
-	ID string
-	BaseUser
-}
-
 type DBUser struct {
 	ID int64 `db:"id"`
 	BaseUser
```
Comment (itaiad200) [inline pkg/auth/model/model.go:106]: I suggest a rename just like with `Group` and `Policy`
```suggestion
	User
```

**Review by itaiad200 — APPROVED** [review]
LGTM with some comments. Please wait for the CN team to review it too

File: pkg/auth/model/model.go [hunk 2]
```diff
@@ -217,23 +201,19 @@ func (s *Statements) Scan(src interface{}) error {
...
-			Source:            pb.Source,
-		},
+func UserFromProto(pb *UserData) *BaseUser {
+	return &BaseUser{
+		CreatedAt:         pb.CreatedAt.AsTime(),
+		Username:          pb.Username,
+		FriendlyName:      &pb.FriendlyName,
+		Email:             &pb.Email,
+		EncryptedPassword: pb.EncryptedPassword,
+		Source:            pb.Source,
```
Comment (ortz) [inline pkg/auth/model/model.go:211]: The property ExternalID is missing here compared to the BaseUser

Comment (N-o-Z) [reply pkg/auth/model/model.go:211]: Nice catch! :)

File: pkg/auth/authenticator.go [hunk 3]
```diff
@@ -218,7 +218,7 @@ func (la *LDAPAuthenticator) AuthenticateUser(ctx context.Context, username, pas
 	user, err := la.AuthService.GetUser(ctx, dn)
 	if err == nil {
 		logger.WithField("user", fmt.Sprintf("%+v", user)).Debug("Got existing user")
-		return user.ID, nil
+		return user.Username, nil
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/auth/authenticator.go:221]: @ortz You are right, we need to change the way we hold the user credentials to keep the username instead/ in addition to the user id and work with username only in the authenticators

**Comment by N-o-Z** [comment]
@guy-har  Would appreciate your review on this

**Review by guy-har — CHANGES_REQUESTED** [review]
Great change :)
Requesting changes due to bug in `middleware.go`
One thing we should be aware of is that updating to this version will log out all users that are currently logged in (maybe, we should just remember to write it down in the changelog).

**Review by guy-har — APPROVED** [review]

**Comment by N-o-Z** [comment]
@guy-har  Thanks! Though I still need to understand why TestMigrate fails (probably because of caching?)

**Pull request merged**